event/oracle/upgrade_oracle_event.go

```diff
@@ -0,0 +1,178 @@
+package oracle
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"github.com/edgelesssys/ego/enclave"
+	"github.com/medibloc/panacea-doracle/sgx"
+	"github.com/tendermint/tendermint/light/provider"
+
+	oracletypes "github.com/medibloc/panacea-core/v2/x/oracle/types"
+	"github.com/medibloc/panacea-doracle/event"
+	"github.com/medibloc/panacea-doracle/panacea"
+	log "github.com/sirupsen/logrus"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+)
+
+type UpgradeOracleEvent struct {
+	reactor event.Reactor
+}
+
+var _ event.Event = (*UpgradeOracleEvent)(nil)
+
+func NewUpgradeOracleEvent(s event.Reactor) UpgradeOracleEvent {
+	return UpgradeOracleEvent{s}
+}
+
+func (e UpgradeOracleEvent) GetEventType() string {
+	return "message"
+}
+
+func (e UpgradeOracleEvent) GetEventAttributeKey() string {
+	return "action"
+}
+
+func (e UpgradeOracleEvent) GetEventAttributeValue() string {
+	return "'OracleUpgrade'"
+}
+
+func (e UpgradeOracleEvent) EventHandler(event ctypes.ResultEvent) error {
+	uniqueID := event.Events[oracletypes.EventTypeUpgradeVote+"."+oracletypes.AttributeKeyUniqueID][0]
+	addressValue := event.Events[oracletypes.EventTypeUpgradeVote+"."+oracletypes.AttributeKeyOracleAddress][0]
+	queryClient := e.reactor.QueryClient()
+
+	oracleRegistration, err := queryClient.GetOracleRegistration(addressValue, uniqueID)
+	if err != nil {
+		log.Infof("failed to get oracleRegistration, voting ignored. uniqueID(%s), address(%s). %v", uniqueID, addressValue, err)
+		return err
+	}
+
+	voteOption, err := e.verifyAndGetVoteOption(oracleRegistration)
+	if err != nil {
+		switch voteOption {
+		case oracletypes.VOTE_OPTION_NO:
+			log.Infof("vote No due to error while verify: %v", err)
+		case oracletypes.VOTE_OPTION_UNSPECIFIED:
+			log.Errorf("can't vote due to error whiile verify. uniqueID(%s), votingTargetAddress(%s), error(%v)",
+				uniqueID,
+				addressValue,
+				err)
+			return err
+		default:
+			log.Warnf("if an error occurs, no other voteOption is possible. uniqueID(%s), votingTargetAddress(%s) voteOption(%s), err(%v)",
+				uniqueID,
+				addressValue,
+				voteOption,
+				err)
```

Comment:
I don't think the case of `default` will happen, shouldn't an error be returned here?

Reply:
Current, we handler only `VOTE_OPTION_NO` and `VOTE_OPTION_UNSPECIFIED`.(Another code)
So I simply want printing log when unexpected set a voteOption.

Umm... Actually, I consider this part whether return error or not.
First of all, I will keep this code.

Reply:
Got it!